test/EndToEndTests/Tests/Client/Microsoft.OData.Client.E2E.Tests/ClientWithoutTypeResolverTests/Tests/JsonWithoutTypeResolverTests.cs

```diff
@@ -0,0 +1,185 @@
+﻿//-----------------------------------------------------------------------------
+// <copyright file="JsonWithoutTypeResolverTests.cs" company=".NET Foundation">
+//      Copyright (c) .NET Foundation and Contributors. All rights reserved.
+//      See License.txt in the project root for license information.
+// </copyright>
+//------------------------------------------------------------------------------
+
+using Microsoft.AspNetCore.OData;
+using Microsoft.AspNetCore.OData.Routing.Controllers;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.OData.Client.E2E.TestCommon;
+using Microsoft.OData.Client.E2E.Tests.ClientWithoutTypeResolverTests.Server;
+using Microsoft.OData.Client.E2E.Tests.Common.Clients.EndToEnd.Default;
+using Microsoft.OData.Client.E2E.Tests.Common.Server.EndToEnd;
+using Xunit;
+
+namespace Microsoft.OData.Client.E2E.Tests.ClientWithoutTypeResolverTests.Tests
+{
+    public class JsonWithoutTypeResolverTests : EndToEndTestBase<JsonWithoutTypeResolverTests.TestsStartup>
+    {
+        private readonly Uri _baseUri;
+        private readonly Container _context;
+        public class TestsStartup : TestStartupBase
+        {
+            public override void ConfigureServices(IServiceCollection services)
+            {
+                services.ConfigureControllers(typeof(JsonWithoutTypeResolverTestsController), typeof(MetadataController));
+
+                services.AddControllers().AddOData(opt => opt.EnableQueryFeatures()
+                    .AddRouteComponents("odata", CommonEndToEndEdmModel.GetEdmModel()));
+            }
+        }
+
+        public JsonWithoutTypeResolverTests(TestWebApplicationFactory<TestsStartup> fixture)
+            : base(fixture)
+        {
+            _baseUri = new Uri(Client.BaseAddress, "odata/");
+            _context = new Container(_baseUri);
+            _context.HttpClientFactory = HttpClientFactory;
+        }
+
+        [Fact]
+        public void DerivedTypeFeedQuery_Executes_Successfully()
+        {
+            var baseQueryResults = _context.Execute<Common.Clients.EndToEnd.PageView>(new Uri(_baseUri.OriginalString + "PageViews/Microsoft.OData.Client.E2E.Tests.Common.Server.EndToEnd.ProductPageView")).ToArray();
+
+            var derivedQueryResults = _context.Execute<Common.Clients.EndToEnd.DiscontinuedProduct>(new Uri(_baseUri.OriginalString + "Products/Microsoft.OData.Client.E2E.Tests.Common.Server.EndToEnd.DiscontinuedProduct")).ToArray();
+
+            Assert.NotNull(baseQueryResults); // Verify that baseQueryResults is not null

```

Comment:
I don't think `ToArray()` can ever return null, so this assertion is unnecessary.